benchmarks/synchrobench/src/main/java/com/yahoo/oak/synchrobench/MyBuffer.java

```diff
@@ -37,70 +36,46 @@ public int compareTo(MyBuffer o) {
         return compareBuffers(this, o);
     }
 
-    public static void serialize(MyBuffer inputBuffer, OakUnsafeDirectBuffer targetBuffer) {
-        serialize(inputBuffer, targetBuffer.getByteBuffer(), targetBuffer.getOffset());
-    }
-
-    public static void serialize(MyBuffer inputBuffer, ByteBuffer targetBuffer, int targetPos) {
+    public static void serialize(MyBuffer inputBuffer, OakScopedWriteBuffer targetBuffer) {
         // In the serialized buffer, the first integer signifies the size.
-        targetBuffer.putInt(targetPos, inputBuffer.capacity);
-        // Thus, the data position starts after the first integer.
-        targetPos += Integer.BYTES;
-
+    	targetBuffer.putInt(0,  inputBuffer.capacity);
         OakIntBufferSerializer.copyBuffer(inputBuffer.buffer, DATA_POS, inputBuffer.capacity / Integer.BYTES,
-                targetBuffer, targetPos);
-    }
-
-    public static MyBuffer deserialize(OakUnsafeDirectBuffer inputBuffer) {
-        return deserialize(inputBuffer.getByteBuffer(), inputBuffer.getOffset());
+        		targetBuffer, Integer.BYTES);
```

Comment:
I would remain with `targetPos` internal variable, initiating it initially to zero, then adding `Integer.BYTES`

Reply:
Same for the `deserialize()` below

Reply:
for the write buffer there is no variable to indicate the current offset

Reply:
lets discuss